support for other platforms and upgrade to Rugpi v0.7

koehlma opened the pull request:
This PR adds support for other platforms than Raspberry Pi and contains necessary changes to work with Rugpi version 0.7.

koehlma committed:
chore: remove dependency on `raspberrypi` and add `essentials` recipe

koehlma committed:
chore: restructure layers

koehlma committed:
chore: instruct users to pin a branch

didier-wenzek commented on recipes/essentials/recipe.toml:
```diff
@@ -0,0 +1,10 @@
+    "tedge-bootstrap",
+    "tedge-firmware-update",
+    "avahi-daemon",
+    "mosquitto",
```
Mosquitto is listed here and in the defaults recipe. Is this an issue?

koehlma replied:
No issue but an oversight. Good catch! I will fix that.

didier-wenzek commented on README.md:
````diff
@@ -6,7 +6,7 @@ To make the recipes and layers available, include the following in your `rugpi-b
 ```toml
 [repositories]
-tedge-rugpi-core = { git = "https://github.com/thin-edge/tedge-rugpi-core.git" }
+tedge-rugpi-core = { git = "https://github.com/thin-edge/tedge-rugpi-core.git", branch = "v0.7" }
````
So the plan is to create a branch "v0.7" is this repository to be used from https://github.com/thin-edge/tedge-rugpi-image?

To be honest, I don't really understand the relationship between these two repositories. And between this PR and that other one: https://github.com/thin-edge/tedge-rugpi-image/pull/79.

koehlma replied:
Indeed, this would be the plan. The idea behind the repositories feature of Rugpi Bakery is to make it easy to share recipes. To make sure that backwards-incompatible changes to recipes do not break people's builds, it is a good idea to implement some versioning. The way I currently do this is to use branches. Alternatively, tags could be used (with the `tag` property). You can also start with a different version number for `tedge-rugpi-core`. I just thought, it would make sense to keep them in sync with Rugpi's versioning for now.

The `tedge-rugpi-image` repository contains a template that people can use to start building their project. The `tedge-rugpi-core` repository contains the recipes (and layers) necessary to install thin-edge.io into an image – it provides a sort of library that is then used by `tedge-rugpi-image`. This PR contains changes to the library and the other PR corresponding changes to the template.

didier-wenzek replied:
Thanks. This is helpful. Things are clearer now. And yes it makes sense to use branches for various core images of thin-edge.

didier-wenzek reviewed (commented):
I would need more time to test these updates by building an image.

koehlma committed:
chore: remove `mosquitto` from the `defaults` recipe

The `mosquitto` recipe is already included in the `essentials` recipe, hence, this dependency was redundant.

koehlma commented:
> I would need more time to test these updates by building an image.

The image repository contains a GitHub Actions pipeline that builds all the images (I also tested that before submitting this PR, it should work). Maybe, first create the `v0.7` branch in this repository based on this PR (without actually changing anything in the `main` branch) and then a separate branch in the image repository to test building the images with the pipeline?

didier-wenzek commented:
> Maybe, first create the `v0.7` branch in this repository based on this PR (without actually changing anything in the `main` branch)

As the idea is to "make sure that backwards-incompatible changes to recipes do not break people's builds", yes this would be good.

reubenmiller commented:
@koehlma Thanks for the PR. I changed the base branch to `v0.7` as suggested.
